influxdb/_dataframe_client.py

```diff
@@ -131,25 +132,26 @@ def write_points(self,
                 protocol=protocol)
             return True
 
-    def query(self, query, chunked=False, database=None):
+    def query(self, query, dropna=True, **kwargs):
         """
         Quering data into a DataFrame.
 
-        :param chunked: [Optional, default=False] True if the data shall be
-            retrieved in chunks, False otherwise.
+        :param query: the actual query string
+        :param dropna: drop columns where all values are missing
+        :param **kwargs: additional parameters for ``InfluxDBClient.query``
 
         """
-        results = super(DataFrameClient, self).query(query, database=database)
+        results = super(DataFrameClient, self).query(query, **kwargs)
```

Comment:
| As for passing inexistent arguments causing a runtime error well, the same can be said for passing inexistent arguments directly to InfluxDBClient.query

The difference though is, without `kwargs` you get warnings from static analyser tools like pylint, flake8 etc, while with `kwargs` code will only break when it tries to execute at run time. That would be fine for internal functions which are only used within library but this is a public endpoint.

Taking that argument to its extreme, why even use named keyword parameters in the first place? Just use `kwargs` everywhere. You could, usage wise will work the same way, but the code becomes unreadable and users don't know what arguments they can pass in without reading documentation or going through code.

It's more code to write and maintain sure, but is better practice and clearer from a user's point of view, meaning looking at method signature tells you its parameters. 

As always though, these things are subject to opinion - you may not agree and that's fine! :) It's really up to maintainers to set guidelines on things like this.